doc/design/doi-generation-2.md

```diff
@@ -0,0 +1,214 @@
+
+## Background
+
+Initially proposed/discussed in
+
+- [Create and maintain a "Findable" DOI for the Dandiset as a whole](https://github.com/dandi/dandi-archive/issues/1319)
+
+and boils down to the adoption of approach of Zenodo of having a DOI which always points to the latest version of the record.
+[Zenodo uses the language](https://support.zenodo.org/help/en-gb/1-upload-deposit/97-what-is-doi-versioning) `Concept DOI` to mean a top-level DOI that references all versions, which we will refer to as `Dandiset DOI`.
+
+DataCite allows for three types of DOIs ([DataCite](https://support.datacite.org/docs/what-does-the-state-of-the-doi-mean)):
+
+- `Draft`. We do not use those.
+  *Can be deleted, and they require only the identifier itself in order to be created or saved. They can be updated to either Registered or Findable DOIs. Registered and Findable DOIs may not be returned to the Draft state, which means that changing the state of a Draft is final.*
+- `Registered`. Like `Findable` but not indexed for search, so we do not use them.
+- `Findable`. Is the type we use for published dandisets.
+  Requires to be valid (pass validation to fit the datacite schema) to be created.
+
+## Proposed Solution
+
+- Upon creation of a **public** dandiset, mint a `Dandiset DOI` (a DataCite `Draft DOI`) `10.48324/dandi.{dandiset.id}` with
+  - *minimal metadata* entered during creation request (title, description, license)
+  - DLP URL `https://dandiarchive.org/dandiset/{dandiset.id}`
+  - For embargoed dandiset, **do not** specify any metadata besides the DLP URL.
+  - If minting a DOI fails, we need to raise exception to inform developers about the issue but proceed with the creation of the dandiset.
+- Upon changes to a non-embargoed, draft dandiset metadata record:
+  - If `Draft DOI`, attempt to "promote" it to `Findable`.
+    - If validation fails - keep `Draft DOI` (very limited validation), attempt to update datacite metadata record while keeping the same target URL.
+    - **Question to clear up**: what happens to `Draft DOI` if metadata record is invalid? It seems to create one with no metadata, but does it update only the fields it knows about?
+  - If `Findable DOI`
+    - if draft version with no prior publications, but which had legit metadata, we try to update metadata.
+      - **Question to clear up** If fails, we either ignore or just add a comment somewhere that "record might not reflect the most recent changes to draft version".
+      - **Question to clear up** If we add to validation procedures to dandiset updates, (validation against datacite metadata record), we can report errors to the user so they can be addressed prior to attempted publication. May be we should validate only if no other errors (our schema validation) were detected to reduce noise, or just give a summary that "Metadata is not satisfying datacite model, fix known metadata errors first."
```

Comment:
@djarecka and I discussed this and we aren't convinced this is really necessary, especially since it leads to that awkward scenario of a Findable DOI with  incorrect metadata

"Adding a comment somewhere" -- Since we cant use the data on the dandiset, (it would be invalid) we could possibly retrieve the DOI, and modify that data, but this feels unclean to me.

IMO the cleaner path would be to simply leave Draft Dandisets as Draft DOIs until their first publish. At that point mint the Dandiset DOI. And only update the Dandiset DOI after further successful publishes.

Reply:
I do not think here it is concerned with Draft-vs-Findable, and rather to validate metadata against datacite for draft version. May be this comment is mis-placed?